contracts/GenericLender/GenericAave.sol

```diff
@@ -28,42 +32,75 @@ contract GenericAave is GenericLenderBase {
 
     IProtocolDataProvider public constant protocolDataProvider = IProtocolDataProvider(address(0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d));
     IAToken public aToken;
+    IStakedAave public constant stkAave = IStakedAave(0x4da27a545c0c5B758a6BA100e3a049001de870f5);
+
+    bool public isIncentivised;
+
+    address public constant WETH =
+        address(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+    address public constant AAVE =
+        address(0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9);
+
+    IUniswapV2Router02 public constant router =
+        IUniswapV2Router02(address(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D));
+
+    uint256 constant internal SECONDS_IN_YEAR = 365 days;
 
     constructor(
         address _strategy,
         string memory name,
-        IAToken _aToken
+        IAToken _aToken,
+        bool _isIncentivised
     ) public GenericLenderBase(_strategy, name) {
-        _initialize(_aToken);
+        _initialize(_aToken, _isIncentivised);
     }
 
-    function initialize(IAToken _aToken) external {
-        _initialize(_aToken);
-    }
-
-    function _initialize(IAToken _aToken) internal {
-        require(address(aToken) == address(0), "GenericAave already initialized");
-
-        aToken = _aToken;
-        require(_lendingPool().getReserveData(address(want)).aTokenAddress == address(_aToken), "WRONG ATOKEN");
-        IERC20(address(want)).safeApprove(address(_lendingPool()), type(uint256).max);
+    function initialize(IAToken _aToken, bool _isIncentivised) external {
+        _initialize(_aToken, _isIncentivised);
     }
 
     function cloneAaveLender(
         address _strategy,
         string memory _name,
-        IAToken _aToken
+        IAToken _aToken,
+        bool _isIncentivised
     ) external returns (address newLender) {
         newLender = _clone(_strategy, _name);
-        GenericAave(newLender).initialize(_aToken);
+        GenericAave(newLender).initialize(_aToken, _isIncentivised);
     }
 
-    function nav() external view override returns (uint256) {
-        return _nav();
+    // for the management to activate / deactivate incentives functionality
+    function setIsIncentivised(bool _isIncentivised) external management {
+        require(address(aToken.getIncentivesController()) != address(0), "!aToken does not have incentives controller set up");
+        isIncentivised = _isIncentivised;
     }
 
-    function _nav() internal view returns (uint256) {
-        return want.balanceOf(address(this)).add(underlyingBalanceStored());
+
+    function withdraw(uint256 amount) external override management returns (uint256) {
+        return _withdraw(amount);
```

Comment:
This seems to be using the same _withdraw method of the original versions, does this need to change accounting for stkAAVE?

Or im trying to picture what is the emergency exit on this strategy, can you detail the steps in case we want to exit as fast as possible?

Reply:
as the workflow for claiming rewards is fully independent from main workflow, the withdraw function works in the same way for incentivised and not incentivised aTokens (and previous version)

for emergency exit, management would call `withdrawAll()` or `removeLender(lenderAddress, true)`(in Strategy)

this would leave rewards behind (which can be safely claimed after calling withdrawAll() and, once they are sold for want, can be sent to strategy by calling withdrawAll() again)